Mathlib/Algebra/PUnitInstances.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2019 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+
+! This file was ported from Lean 3 source module algebra.punit_instances
+! leanprover-community/mathlib commit 6cb77a8eaff0ddd100e87b1591c6d3ad319514ff
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Module.Basic
+import Mathlib.Algebra.GCDMonoid.Basic
+import Mathlib.Algebra.GroupRingAction.Basic
+import Mathlib.GroupTheory.GroupAction.Defs
+import Mathlib.Order.CompleteBooleanAlgebra
+
+/-!
+# Instances on PUnit
+
+This file collects facts about algebraic structures on the one-element type, e.g. that it is a
+commutative ring.
+-/
+
+
+universe u
+
+namespace PUnit
+
+variable {R S : Type _} (x y : PUnit.{u + 1}) (s : Set PUnit.{u + 1})
+
+@[to_additive]
+instance commGroup: CommGroup PUnit where
+  mul := fun _ _ => PUnit.unit
```

Comment:
Does `unit` work instead of `PUnit.unit`?  You're in the `PUnit` namespace.  Same in many places throughout.

Reply:
done